File: BegginersSelection/ABC083B.rb
def sum_digit(i, a, b)
    # 各桁の総和
    sum = 0
    while(i > 0) do
        sum += i % 10
        i /= 10
    end
    if a <= sum && sum <= b
        return true
    else 
        return false
    end
end

line = gets.chomp.split(" ").map(&:to_i)
n = line[0]
a = line[1]
b = line[2]
ans = 0

(1..n).each do |i|
    if sum_digit(i, a, b)
        ans += i
    end
end
puts ans

File: BegginersSelection/Golang/ABC083B.go
package main

import "fmt"

func main() {
	var n, a, b int
	fmt.Scan(&n, &a, &b)
	ans := 0
	for i := 1; i <= n; i++ {
		sum := 0
		for j := i; j > 0; j /= 10 {
			sum += j % 10
		}
		if a <= sum && sum <= b {
			ans += i
		}
	}
	fmt.Println(ans)
}
